subsys/net/lib/lwm2m/lwm2m_registry.c

```diff
@@ -40,6 +40,8 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
 #define BINDING_OPT_MAX_LEN 3 /* "UQ" */
 #define QUEUE_OPT_MAX_LEN   2 /* "Q" */
 
+static K_MUTEX_DEFINE(readwrite_lock);
+static K_MUTEX_DEFINE(registry_lock);
```

Comment:
I don't understand, why we need two locks in one registry.
I think it should be fine to have both use the same mutex.

After all creating and deleting objects is a corner case that rarely happens. I don't see the point of optimizing it.